apollo-router/src/spec/query/change.rs

```diff
@@ -0,0 +1,611 @@
+use std::collections::HashMap;
```

Comment:
This all feels like it might be something that is generally useful within `apollo-rs`. It's fine to have it in the router right now, but consider adding to appropriate `apollo-rs` crate at some point.

Reply:
yes, it will be generally useful elsewhere. Probably in federation-next